bentoml/_internal/types.py

```diff
@@ -150,7 +156,7 @@ def __init__(
             else:
                 self.qualname: str = getattr(module_or_cls, "__name__")
 
-    def __instancecheck__(self, obj) -> "t.TypeGuard[T]":
```

Comment:
can you explain more about this change?

Reply:
Assuming you figured this one out because you marked it as resolved, but for others: pyright complains about the type not matching the `type` superclass, and I think it's a type guard by default since we're now inheriting `type[T]`.

Reply:
So this change is invoked because of the previous usage change from `lazy_type.isinstance(obj)` to `isinstance(obj, lazy_type)`?

Reply:
@sauyon On my side this change breaks type guard. It can not narrow the unions down anymore.
How about just keeping using the original solution?